### PR TITLE
Change GPL to apache 2.0

### PR DESCRIPTION
**Pull request by leeyr338**
Change GPL to apache 2.0.

* Change LICENSE from GPL to Apache 2.0.
* Add license announce to Cargo.toml.
* Add license announce to each rust file.

### CODE REVIEW

File: cita-chain/types/src/receipt.rs [hunk 1]
```diff
@@ -1,18 +1,16 @@
-// Copyright 2015-2017 Parity Technologies (UK) Ltd.
```
Comment (kaikai1024) [inline cita-chain/types/src/receipt.rs]: Merge this PR after this file is replaced?

Comment (leeyr338) [reply cita-chain/types/src/receipt.rs]: You mean we can not change this file to apache license ?

Comment (leeyr338) [reply cita-chain/types/src/receipt.rs]: You mean we can not change this file to apache license for now ?

Comment (kaikai1024) [reply cita-chain/types/src/receipt.rs]: Not Professional.
The project muse be GPL even there is only one GPL file accoring the GPL license.
Just an advice, yes, we can. but i can not apporve.

Comment (leeyr338) [reply cita-chain/types/src/receipt.rs]: > The project muse be GPL even there is only one GPL file accoring to the GPL license.

Yes!

**Pull request merged**